plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ConnectionWrapper.java

```diff
@@ -192,7 +196,13 @@ void connectionChanged(String property, Object oldValue, Object newValue) {
 
 	private ProjectWrapper findProjectWrapper(IResource resource) {
 		synchronized (projects) {
-			return projects.get(resource.getProject());
+			// `projects.get(resource.getProject())` was replaced, because project

```

Comment:
This was a convenience method that made an additional call to get the project that owns a resource.  It probably should be deprecated and what you have here is a better implementation.